### PR TITLE
GH2119: Added missing .NET Core publish parameters

### LINKED ISSUE #2119: DotNetCore Publish misses Force / Self contained / Sources settings

### PR DESCRIPTION
**Pull request by devlead**
Missing DotNetCorePublishSettings added

* Force
* SelfContained
* Sources
* Fixes #2119

### CODE REVIEW

File: src/Cake.Common/Tools/DotNetCore/Publish/DotNetCorePublishSettings.cs [hunk 1]
```diff
@@ -54,6 +55,30 @@ public sealed class DotNetCorePublishSettings : DotNetCoreSettings
...
+        /// </remarks>
+        public bool SelfContained { get; set; }
+
+        /// <summary>
+        /// Gets or sets the specified NuGet package sources to use during the restore.
+        /// </summary>
+        /// <remarks>
+        /// Requires .NET Core 2.x or newer.
+        /// </remarks>
+        public ICollection<string> Sources { get; set; }
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/DotNetCore/Publish/DotNetCorePublishSettings.cs]: I think it would be a good idea to initialize this in the constructor to avoid null reference exceptions if the user assumes it's not null.

Comment (devlead) [reply src/Cake.Common/Tools/DotNetCore/Publish/DotNetCorePublishSettings.cs]: That would make it inconsistent with i.e.
https://github.com/cake-build/cake/blob/develop/src/Cake.Common/Tools/DotNetCore/Restore/DotNetCoreRestoreSettings.cs#L19

https://github.com/cake-build/cake/blob/develop/src/Cake.Common/Tools/NuGet/Restore/NugetRestoreSettings.cs#L32

though.

Comment (patriksvensson) [reply src/Cake.Common/Tools/DotNetCore/Publish/DotNetCorePublishSettings.cs]: But not inconsistent with https://github.com/cake-build/cake/blob/develop/src/Cake.Common/Tools/MSBuild/MSBuildSettings.cs#L149

File: tests/integration/Cake.Core/Scripting/AddinDirective.cake [hunk 2]
```diff
@@ -11,7 +11,13 @@ Task("Cake.Core.Scripting.AddinDirective.LoadNetStandardAddin")
...
     ";
 
-    CakeExecuteExpression(script);
+    CakeExecuteExpression(script,
+    	new CakeSettings {
+    		EnvironmentVariables = new Dictionary<string, string>{
+    			{"CAKE_NUGET_SOURCE", $"{Paths.Temp}/tools/Addins"},
+    			{"CAKE_PATHS_TOOLS", $"{Paths.Temp}/tools"},
+    			{"CAKE_PATHS_MODULES", $"{Paths.Temp}/tools/Modules"}
+    		}});
```
Comment (mholo65) [inline tests/integration/Cake.Core/Scripting/AddinDirective.cake]: Why was this changed?

File: tests/integration/Cake.Core/Scripting/AddinDirective.cake [hunk 3]
```diff
@@ -11,7 +11,13 @@ Task("Cake.Core.Scripting.AddinDirective.LoadNetStandardAddin")
         Information(""The answer to life: {0}"", TheAnswerToLife);
     ";
 
-    CakeExecuteExpression(script);
+    CakeExecuteExpression(script,
+    	new CakeSettings {
+    		EnvironmentVariables = new Dictionary<string, string>{
+    			{"CAKE_PATHS_ADDINS", $"{Paths.Temp}/tools/Addins"},
```
Comment (devlead) [inline tests/integration/Cake.Core/Scripting/AddinDirective.cake]: @mholo65 this was changed because saw an issue on multiple Windows systems with addin installation as Cake changes current directory to script dir, which for expression is Windows temp director, setting it to be in a known directory within integration tests sorted it.

Long term CakeExecuteExpression probably should set these by default if not already set to the paths in current cake process config as that's probably the expected behaviour. 

Unreleated to issue so could remove, just couldn't run integration tests locally without it atm.

**Comment by mholo65** [comment]
@devlead, your changes have been merged. Thank you for your contribution.
